src/library/attribute_manager.h

```diff
@@ -155,20 +155,31 @@ template class typed_attribute<indices_attribute_data>;
 /** \brief Attribute that represents a list of indices. input and output are 1-indexed for convenience. */
 typedef typed_attribute<indices_attribute_data> indices_attribute;
 
-void register_attribute(attribute_ptr);
-attribute const & get_attribute(name const & attr);
-void get_attributes(buffer<attribute const *> &);
+class user_attribute_ext {
+public:
+    virtual name_map<attribute_ptr> const & get_attributes(environment const & env);
+    virtual void write_entry(serializer &, attr_data const &) {}
+    virtual attr_data_ptr read_entry(deserializer &) {
+        return attr_data_ptr(new attr_data);
+    }
+};
```

Comment:
@leodemoura I made this into a class after I noticed that I needed multiple callbacks